style: Prepare introduction of `ruff` and `ruff format`

echoix opened the pull request:
I'm preparing the repo to use `ruff` and `ruff format`, like the main repo.

Most of the changes here are to apply non-conflicting changes, in other words, `ruff format` changes that are not reverted by the current black style. It will make reviewing the next PRs easier.

echoix committed:
style: Configure black to also target python 3.13

echoix committed:
style: Add basic metadata to pyproject.toml to make ruff work nicely

echoix committed:
style: Apply `ruff format` changes not conflicting with Black 25

cwhite911 reviewed (approved):
Cool! Didn't realized we switched from Black.

wenzeslaus reviewed (approved):
I checked just couple files at random while scrolling through it, but it looks good.

echoix commented:
> I checked just couple files at random while scrolling through it, but it looks good.

There were all correct, except 2 or 3 places. 1 had a implicit string concatenation merged together and it had an extra space between sentences that I removed, and 1 or 2 had a comma after the print, making a tuple, similar to what we fixed in the main repo yesterday/today.

echoix commented:
> Cool! Didn't realized we switched from Black.

It was discussed since a month, but it was implemented this weekend and merged this afternoon. So you're not late on the news haha
